### PR TITLE
feat: switch from http to ws connection (eth execution client)

### PR DESCRIPTION
**Pull request by Vid201**
Some other things in this PR:

- switch from anyhow to eyre
- update bundler spec tests to latest commit
- update geth version from to v1.12.0

### CODE REVIEW

**Review by zsluedem — COMMENTED** [review]
What if the websocket connection is lost  because of network issue? Do you have plans to address that?Does ethers ws handle that internally?

The problem with websocket is that we need to provide a way when the connection is lost. If we don't have that, when the situation happens, the bundler would just crash and got no more data from websocket. As for normal http request, if one of the http request is lost, it would just got one task crash(maybe this task would crash the whole bundler. It is possible. But http leave codes easier to write)

**Comment by Vid201** [comment]
> What if the websocket connection is lost because of network issue? Do you have plans to address that?Does ethers ws handle that internally?
> 
> The problem with websocket is that we need to provide a way when the connection is lost. If we don't have that, when the situation happens, the bundler would just crash and got no more data from websocket. As for normal http request, if one of the http request is lost, it would just got one task crash(maybe this task would crash the whole bundler. It is possible. But http leave codes easier to write)

Yes, that is the problem that still needs to be addressed in this PR. At the moment, if the connection is dropped, the bundler doesn't crash (same as HTTP). But if the eth client is up again, the connection is not reestablished. I am planning to make a wrapper around eth_client, that offers a way to do healthcheck and connect if connection is not live. Then the bundler pings eth_client every x seconds. What do you think?

I think websockets will be useful when we subscribe to blocks to check which user operations are included onchain.

File: bin/silius/src/cli/args.rs [hunk 1]
```diff
@@ -172,6 +172,10 @@ pub struct RpcArgs {
     /// By default, this option is set to `*`.
     #[clap(long = "ws.origins", value_delimiter = ',', default_value = "*")]
     pub ws_origins: Vec<String>,
+
+    /// Ethereum execution client proxy HTTP RPC endpoint
+    #[clap(long)]
+    pub eth_client_proxy_address: Option<String>,
```
[the remark this answers is not in the document]
Comment (Vid201) [reply bin/silius/src/cli/args.rs:178]: Hmm, I am not sure what you mean by that

**Comment by Vid201** [comment]
> > What if the websocket connection is lost because of network issue? Do you have plans to address that?Does ethers ws handle that internally?
> > The problem with websocket is that we need to provide a way when the connection is lost. If we don't have that, when the situation happens, the bundler would just crash and got no more data from websocket. As for normal http request, if one of the http request is lost, it would just got one task crash(maybe this task would crash the whole bundler. It is possible. But http leave codes easier to write)
> 
> Yes, that is the problem that still needs to be addressed in this PR. At the moment, if the connection is dropped, the bundler doesn't crash (same as HTTP). But if the eth client is up again, the connection is not reestablished. I am planning to make a wrapper around eth_client, that offers a way to do healthcheck and connect if connection is not live. Then the bundler pings eth_client every x seconds. What do you think?
> 
> I think websockets will be useful when we subscribe to blocks to check which user operations are included onchain.

@zsluedem I fixed the reconnection problems (https://github.com/Vid201/silius/blob/feat/ws/Cargo.toml#L43). This is now handled in ethers-rs internally (https://github.com/gakonst/ethers-rs/commit/d9c2bafd0c462a5b009fb1712c6cc5c866e0ef73). When the connection is lost, it tries to reconnect every 2 seconds. When the node becomes online again, it works normally onwards. The behavior is similar now to lost http requests.

**Pull request merged**